src/runner.py

```diff
@@ -122,6 +128,10 @@ def __init__(
 
         self._shared_fs: Optional[shared_fs.SharedFilesystem] = None
 
+        if aproxy_address := self.config.proxies.get("aproxy_address"):
+            # If aproxy is used, the proxy environment variables are not necessary.
+            self.config.proxies = ProxySetting(aproxy_address=aproxy_address)
```

Comment:
I'm guessing that this has the side-effect of not setting the proxy environment variables? It is a bit confusing that in order to remove proxy settings we set something else. Perhaps take a look at the code to see whether we can make the decision of what environment variables we set for the proxy a bit more explicit?

Reply:
I do a reset here to avoid mutating the original dict, as it could be reused anywhere in the code. But I'll update the code to avoid confusion.